homework/list.js

```diff
@@ -2,78 +2,94 @@
 
 (function() {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+  var NoteListManager = function() {
+  // Local data storage; should sync up with the server.
+    this._listNoteContent = [];
+    // Will be an element as the list wrapper.
+    this._wrapper = document.querySelector('#note-list-wrapper');
+  };
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+  NoteListManager.prototype = {
+    start() {      
+      this.fetchList((function(data) {
```

Comment:
Please, "Promisify" this method and then use it as

    this.fetchList.then(() => {
        this.updateList();
        //....
    });